python/cudf/cudf/core/buffer.py

```diff
@@ -1,218 +1,61 @@
-        self.capacity = capacity
-        self.dtype = self.mem.dtype
-
-    def serialize(self):
-        """Called when dask.distributed is performing a serialization on this
-        object.
-
-        Do not use this directly.  It is invoked by dask.distributed.
-
-        Parameters
-        ----------
-        serialize : callable
-             Used to serialize data that needs serialization .
-        context : dict; optional
-            If not ``None``, it contains information about the destination.
-
-        Returns
-        -------
-        (header, frames)
-            See custom serialization documentation in dask.distributed.
-        """
-        header = {}
-        header["type"] = pickle.dumps(type(self))
-        header["cuda_array_interface"] = self.mem.__cuda_array_interface__
-        return header, [self.mem]
-
-    @classmethod
-    def deserialize(cls, header, frames):
-        """Called when dask.distributed is performing a deserialization for
-        data of this class.
-
-        Do not use this directly.  It is invoked by dask.distributed.
-
-        Parameters
-        ----------
-
-        deserialize : callable
-             Used to deserialize data that needs further deserialization .
-        header, frames : dict
-            See custom serialization documentation in dask.distributed.
-
-        Returns
-        -------
-        obj : Buffer
-            Returns an instance of Buffer.
-        """
-        assert len(frames) == 1, "Use Buffer.serialize() for serialization"
-        iface = header["cuda_array_interface"]
-        if (
-            len(iface["shape"]) == 0 or 0 in iface["shape"]
-        ):  # The array is empty
-            arr = rmm.device_array(
-                iface["shape"], dtype=np.dtype(iface["typestr"])
-            )
+class Buffer:
+    def __init__(self, ptr, size=None, owner=None):
+        if not ptr:
+            self.ptr = 0
+            self.size = 0
         else:
-            # Updating the data pointer to the frame data.
-            iface["data"] = frames[0].__cuda_array_interface__["data"]
-
-            # We need an object that exposes __cuda_array_interface__
-            _dummy_iface = types.SimpleNamespace()
-            _dummy_iface.__cuda_array_interface__ = iface
-
-            # Allocating a new RMM CUDA array with shape and dtype as specified
-            # in `iface` and copy the frame data to it, which makes the memory
-            # survive until the new Buffer goes out of scope.
-            # TODO: when DASK supports RMM, we can simply return it as is.
-            arr, _ = rmm.auto_device(_dummy_iface)
-        return Buffer(arr)
+            self.ptr = ptr
+            self.size = size
+        self._owner = owner
 
     def __reduce__(self):
-        cpumem = self.to_array()
-        # Note: pickled Buffer only stores *size* element.
-        return type(self), (cpumem,)
+        return self.__class__.from_array_like, (self.to_host_array(),)
```

Comment:
Is there a way to copy to pinned host memory here?

Reply:
Future optimization in rmm is needed. Resolving for now.